### PR TITLE
CVSL-1741 add endpoint to get licences for approval

### PR DESCRIPTION
**Pull request by gurps-singh**
This PR is to introduce a new endpoint to retrieve the licences which need approving for the CA. This replaces the call in the frontend for `getLicencesForApproval`.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/ApprovedLicenceSummary.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceKind
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType
+import java.time.LocalDateTime
+
```
Comment (gurps-singh) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/ApprovedLicenceSummary.kt:9]: This currently matches what we have as a licence in `ManagedCase` in `server/@types/managedCase.ts` in the frontend with the addition of the CRN. Not included any release date (conditional release date) in particular which I am still a bit on the fence on as the results are sorted by the conditional release date but we are not sending that back as part of the response?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/request/ApproveLicencesRequest.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request
+
+import io.swagger.v3.oas.annotations.media.Schema
+
```
Comment (gurps-singh) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/request/ApproveLicencesRequest.kt:4]: Looking at the `getLicencesForApproval` call on the frontend, the only thing we require are the `filteredPrisons` as the other values are either null or can be plugged straight into the query to retrieve the relevant licences (see `LicenceRepository` and `LicenceService` changes).

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/ApprovedLicenceSummary.kt (hunk 1 above)
Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/ApprovedLicenceSummary.kt:9]: there's a risk that there were only enough things in that licence type that were required for compilation and that the nunjucks template referred to more. 

The type isn't very good - should really expand or refer to existing types

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/request/ApproveLicencesRequest.kt (hunk 2 above)
Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/request/ApproveLicencesRequest.kt:4]: Yes, that makes sense

**Review by andrewrlee — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/ApprovedLicenceSummary.kt (hunk 1 above)
Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/ApprovedLicenceSummary.kt:9]: As endpoint is not currently being used, will add changes to separate PR.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**